src/openrct2/ride/TrackData.cpp

```diff
@@ -5428,6 +5428,745 @@ constexpr static uint8_t TrackTypeToSpinFunction[TrackElemType::Count] = {
     NO_SPIN, NO_SPIN, NO_SPIN, NO_SPIN, NO_SPIN, NO_SPIN, NO_SPIN, NO_SPIN, NO_SPIN
 };
 
+template<int32_t TConstant> static int32_t EvaluatorConst(const int16_t)
+{
+    return TConstant;
+}
+
+static int32_t EvaluatorVerticalLoop(const int16_t progress)
+{
+    return (abs(progress - 155) / 2) + 28;
+}
+
+static int32_t EvaluatorHalfLoopUp(const int16_t progress)
+{
+    return ((static_cast<uint16_t>(-(progress - 155))) / 2) + 28;
+}
+
+static int32_t EvaluatorHalfLoopDown(const int16_t progress)
+{
+    return (progress / 2) + 28;
+}
+
+static int32_t EvaluatorWaterSplash(const int16_t progress)
+{
+    if (progress < 32)
+        return -150;
+    if (progress < 64)
+        return 150;
+    if (progress < 96)
+        return 0;
+    if (progress < 128)
+        return 150;
+    return -150;
```

Comment:
```suggestion
    if (progress < 32)
        return -150;
    else if (progress < 64)
        return 150;
    else if (progress < 96)
        return 0;
    else if (progress < 128)
        return 150;
    else 
        return -150;
```

I wondered if this makes it any more efficient for the compiler, but it seems like it doesn't. Maybe it makes things a bit clearer?

Reply:
These will both compile to the same instructions